edit-post/api/plugin.js

```diff
@@ -0,0 +1,87 @@
+/* eslint no-console: [ 'error', { allow: [ 'error' ] } ] */
+
+/**
+ * WordPress dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { Component } from '@wordpress/element';
+
+/* External dependencies */
+import { isFunction, map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { PluginContextProvider, withPluginContext } from './components/context';
+
+const plugins = {};
+
+/**
+ * Registers a plugin to the editor.
+ *
+ * @param {Object}   settings        The settings for this plugin.
+ * @param {string}   settings.name   The name of the plugin.
+ * @param {Function} settings.render The function that renders the plugin.
+ *
+ * @return {Object} The final plugin settings object.
+ */
+function registerPlugin( settings ) {
+	if ( typeof settings !== 'object' ) {
+		console.error(
+			'No settings object provided!'
+		);
+		return null;
+	}
+	if ( typeof settings.name !== 'string' ) {
+		console.error(
+			'Plugin names must be strings.'
+		);
+		return null;
+	}
+	if ( ! /^[a-z][a-z0-9-]*$/.test( settings.name ) ) {
+		console.error(
+			'Plugin names must include only lowercase alphanumeric characters or dashes, and start with a letter. Example: "my-plugin".'
+		);
+		return null;
+	}
+	if ( plugins[ settings.name ] ) {
+		console.error(
+			`Plugin "${ settings.name }" is already registered.`
+		);
+	}
+	if ( ! isFunction( settings.render ) ) {
+		console.error(
+			'The "render" property must be specified and must be a valid function.'
+		);
+		return null;
+	}
+
+	settings = applyFilters( 'editPost.registerPlugin', settings, settings.name );
```

Comment:
I don't think it is very useful at the moment as there are only 2 properties `name` and `render`. We can leave it you think that there is value in it, but it should be moved to the beginning of the plugin. I'm also not sure if changing the name of the plugin is a good idea because it might break activation logic.

Reply:
@aduth set the opposite in a previous PR, that the filters should be applied after the plugin has been validated.

Reply:
You can override those properties to any random value in that case. Why wpuld we care only about the original params passed but not about those set after all filters got applied?

Reply:
Because a plugin should at the very least be valid? :)

Reply:
Try this:
```js
wp.hooks.addFilter( 'editPost.registerPlugin', 'my-unique-name', function() {
    return 'Whatever;
} );
```
and let me know if it is going to work properly with your plugin :)

Reply:
>@aduth said the opposite in a previous PR, that the filters should be applied after the plugin has been validated.

I did?